### PR TITLE
proofpoint: fix fetch incidents

### PR DESCRIPTION
**Pull request by jochman**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/19455

## Screenshots
Paste here any images that will help the reviewer

## Does it break backward compatibility?
   - No

## Must have
- [X] Tests

### CODE REVIEW

**Comment by jochman** [comment]
@kirbles19 need to work only on the changelog

**Comment by kirbles19** [comment]
@jochman Done.

File: Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py [hunk 1]
```diff
@@ -157,19 +157,19 @@ def get_events_command(client, args):
...
+    if not last_fetch:
         last_fetch, _ = parse_date_range(first_fetch_time, date_format=DATE_FORMAT, utc=True)
-
-    incidents = []
+    incidents: list = []
     fetch_times = get_fetch_times(last_fetch)
     fetch_time_count = len(fetch_times)
     for index, fetch_time in enumerate(fetch_times):
-
+        if index > 5:
```
Comment (anara123) [inline Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py]: remove this

File: Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py [hunk 2]
```diff
@@ -243,11 +256,19 @@ def fetch_incidents(client, last_run, first_fetch_time, event_type_filter, threa
...
             if raw_event["threatTime"] > fetch_time:
                 last_fetch = raw_event["threatTime"]
-
+            incident['Occurred'] = raw_event["clickTime"] if raw_event["clickTime"] > raw_event["threatTime"] else \
+                raw_event["threatTime"]
             incidents.append(incident)
-    last_fetch_datetime = get_now()
-    next_run = {'last_fetch': last_fetch_datetime}
 
+    last_fetch = last_fetch[:-5] + 'Z' if last_fetch[-5] == '.' else last_fetch
```
Comment (anara123) [inline Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py]: add comment here explaining what are you doing here and why

File: Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py [hunk 3]
```diff
@@ -227,10 +238,12 @@ def fetch_incidents(client, last_run, first_fetch_time, event_type_filter, threa
 
             if raw_event["threatTime"] > last_fetch:
                 last_fetch = raw_event["threatTime"]
-
+            incident['Occurred'] = raw_event["clickTime"] if raw_event["clickTime"] > raw_event["threatTime"] else \
```
Comment (anara123) [inline Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py]: all occurred should be lower cased `occurred`

File: Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py [hunk 4]
```diff
@@ -243,11 +256,19 @@ def fetch_incidents(client, last_run, first_fetch_time, event_type_filter, threa
...
 
+    last_fetch = last_fetch[:-5] + 'Z' if last_fetch[-5] == '.' else last_fetch
+    last_fetch_datetime = datetime.strptime(last_fetch, DATE_FORMAT)
+    last_fetch = (last_fetch_datetime + timedelta(seconds=1)).strftime(DATE_FORMAT)
+    next_run = {'last_fetch': last_fetch}
+
+    # limit incidents to the limit given
+    incidents.sort(key=lambda a: a.get('Occurred'))
+    if len(incidents) > limit:
+        incidents = incidents[:limit]
```
Comment (anara123) [inline Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py]: the last_fetch must be the last_fetch of the 50th incident

Comment (anara123) [reply Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py]: so you should delete all the places where you set the last_Fetch and take the last_fetch from the `occurred` field of an last (50th ) incident

**Pull request merged**